### PR TITLE
Add TaxBenefitSystem doc to the reference doc

### PR DESCRIPTION
**Pull request by fpagnoux**
See http://openfisca.readthedocs.io/en/doc/tax-benefit-system.html

### CODE REVIEW

**Comment by benjello** [comment]
Nice work !

File: openfisca_core/taxbenefitsystems.py [hunk 1]
```diff
@@ -130,13 +136,32 @@ def load_variable(self, variable_class, update = False):
...
+
+    def update_variable(self, variable):
+        """
+        Replaces an existing openfisca variable in the tax benefit system by a new one.
+
+        The new variable must have the same name than the old one.
+
+        If no variable with the given name exists in the tax benefit system, no error will be raised and the variable will be simply added.
+
+        :param variable: Variable to add. Must be a sublcass of Variable or DatedVariable.
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: typo: sublcass => subclass

File: openfisca_core/taxbenefitsystems.py [hunk 2]
```diff
@@ -130,13 +136,32 @@ def load_variable(self, variable_class, update = False):
...
 
-    def add_variable(self, variable_class):
-        return self.load_variable(variable_class, update = False)
+    def add_variable(self, variable):
+        """
+        Adds an openfisca variable to the tax benefit system.
 
-    def update_variable(self, variable_class):
-        return self.load_variable(variable_class, update = True)
+        :param variable: The variable to add. Must be a sublcass of Variable or DatedVariable.
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: typo: sublcass

File: openfisca_core/taxbenefitsystems.py [hunk 3]
```diff
@@ -161,10 +189,16 @@ def add_variables_from_directory(self, directory):
             self.add_variables_from_directory(subdirectory)
 
     def add_variables(self, *variables):
+        """
+        Does the same than :any:`add_variable`, but can take several variables.
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: I would put the whole sentence:
```
Adds a list of OpenFisca `Variable`s to the `TaxBenefitSystem`.

See also: `add_variable`
```

I'm not sure of the syntax for "see also" to be a reference

File: openfisca_core/taxbenefitsystems.py [hunk 4]
```diff
@@ -202,6 +236,9 @@ def neutralize_column(self, column_name):
         self.update_column(column_name, neutralize_column(self.get_column(column_name)))
 
     def add_legislation_params(self, path_to_xml_file, path_in_legislation_tree = None):
+        """
+        Adds an xml file to the legislation parameters.
```
Comment (cbenz) [inline openfisca_core/taxbenefitsystems.py]: xml => XML

Tell path_in_legislation_tree is a list of names describing the destination node of the legislation tree

**Pull request merged**